featuretools/tests/entityset_tests/test_ww_es.py

```diff
@@ -0,0 +1,619 @@
+    dates_df.ww.init(index='backwards_order', time_index='random_order',
+                     logical_types={'random_order': 'Integer', 'special': 'Double'})
+    es = EntitySet('es')
+
+    # Both random_order and special are numeric, but they are different logical types
+    es.add_dataframe('dates_table', dates_df, secondary_time_index={'special': ['dates_backwards']})
+
+    assert es['dates_table'].ww.logical_types['random_order'] == Integer
+    assert es['dates_table'].ww.logical_types['special'] == Double
+
+    assert es['dates_table'].ww.time_index == 'random_order'
+    assert 'special' in es['dates_table'].ww.metadata['secondary_time_index']
+
+
+def test_normalize_dataframe():
+    df = pd.DataFrame({
+        'id': range(4),
+        'full_name': ['Mr. John Doe', 'Doe, Mrs. Jane', 'James Brown', 'Ms. Paige Turner'],
+        'email': ['john.smith@example.com', np.nan, 'team@featuretools.com', 'junk@example.com'],
+        'phone_number': ['5555555555', '555-555-5555', '1-(555)-555-5555', '555-555-5555'],
+        'age': pd.Series([33, None, 33, 57], dtype='Int64'),
+        'signup_date': [pd.to_datetime('2020-09-01')] * 4,
+        'is_registered': pd.Series([True, False, True, None], dtype='boolean'),
+    })
+
+    df.ww.init(index='id', time_index='signup_date')
+    es = EntitySet('es')
+    es.add_dataframe('first_table', df)
+    es.normalize_dataframe('first_table', 'second_table', 'age',
+                           additional_columns=['phone_number', 'full_name'],
+                           make_time_index=True)
+    assert len(es.dataframe_dict) == 2
+    assert 'foreign_key' in es['first_table'].ww.semantic_tags['age']
+
+
+def test_update_dataframe():
+    df = pd.DataFrame({
+        'id': range(4),
+        'full_name': ['Mr. John Doe', 'Doe, Mrs. Jane', 'James Brown', 'Ms. Paige Turner'],
+        'email': ['john.smith@example.com', np.nan, 'team@featuretools.com', 'junk@example.com'],
+        'phone_number': ['5555555555', '555-555-5555', '1-(555)-555-5555', '555-555-5555'],
+        'age': pd.Series([33, None, 33, 57], dtype='Int64'),
+        'signup_date': [pd.to_datetime('2020-09-01')] * 4,
+        'is_registered': pd.Series([True, False, True, None], dtype='boolean'),
+    })
+
+    df.ww.init(index='id')
+    es = EntitySet('es')
+    es.add_dataframe('table', df)
+    original_schema = es['table'].ww.schema
+
+    new_df = df.iloc[2:]
+    es.update_dataframe('table', new_df)
+
+    assert len(es['table']) == 2
+    assert es['table'].ww.schema == original_schema
+
+
+def test_add_last_time_indexes(es):
+    es.add_last_time_indexes(['products'])
+
+    assert 'last_time_index' in es['products'].ww.metadata
+
+
+def test_conflicting_dataframe_names(es):
+    new_es = EntitySet()
+
+    sessions_df = es['sessions'].ww.copy()
+
+    assert sessions_df.ww.name == 'sessions'
+
+    new_es.add_dataframe('different_name', sessions_df)
+    assert sessions_df.ww.name == 'different_name'
```

Comment:
thoughts on warning the user we're changing the name if they supply a dataframe initialized with a different woodwork name than the name arg they supplied?

thoughts on name being optional if supplying a ww name and we infer from the `ww.name` if so?

Reply:
I like the idea of raising a warning if we're changing the name.

For the optional name parameter, I think it could make sense to have the name parameter be optional for a woodwork dataframe that's passed in but still take precedence if it is specified and is different from the one set on the Woodwork dataframe.

Reply:
> For the optional name parameter, I think it could make sense to have the name parameter be optional for a woodwork dataframe that's passed in but still take precedence if it is specified and is different from the one set on the Woodwork dataframe.

for the other arguments that can be set on an initialized woodwork dataframe, we ignore the optional parameters in favor of the having the user configure them on the woodwork dataframe.  would `name` be the sole exception to this if allowed the user param to take precedent?

Reply:
Currently it is the one exception. If we want to ignore it like the rest of the parameters, then we need to make the `name` attribute required like we do with `index`.

I think I'd prefer this over changing the name if it doesn't match and raising a warning. The main difference from the index behavior is that for normal dataframes, if no index is specified, we use the first column. In this case, we would probably just tell users that they have to specify a name when woodwork is not already initialized.

Reply:
sounds good to me 👍

Reply:
Implemented these changes. The one area that I thought might still need some consideration is how we want to handle users passing in a woodwork-initialized DataFrame at EntitySet.__init__. Here's the current behavior:

```python
    df.ww.init(index='id', name='table')
    warning = 'A Woodwork-initialized DataFrame was provided, so the following parameters were ignored: dataframe_name'
    with pytest.warns(UserWarning, match=warning):
        es = EntitySet('es', dataframes={'table': (df,)})
```

I'm not entirely happy with this, because it feels like we shouldn't get any sort of warning if the name in the dataframes dictionary matches the name of the DataFrame. 

My suggestion is this, I think:
1. Only raise the ignored warning in `add_Dataframe` if the name parameter doesn't match the dataframe's name
2. If there is a conflict, warn (from the __init__ method) and go with the dictionary value (the logic being that they chose that name the most recently, in theory. but idk)

Reply:
Added the change where we only raise the ignored warning when the name does match, but error if the names do not match.